pypxe/dhcp.py

```diff
@@ -37,30 +37,43 @@ def __init__(self, **serverSettings):
         self.magic = struct.pack('!I', 0x63825363) #magic cookie
 
+        if self.logger == None:
+            import logging
+            import logging.handlers
+            # setup logger
+            self.logger = logging.getLogger("dhcp")
+            handler = logging.StreamHandler()
+            formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+            handler.setFormatter(formatter)
+            self.logger.addHandler(handler)
+
+            if self.mode_debug:
+                self.logger.setLevel(logging.DEBUG)
+
         if self.http and not self.ipxe:
-            print '\nWARNING: HTTP selected but iPXE disabled. PXE ROM must support HTTP requests.\n'
+            self.logger.warning('HTTP selected but iPXE disabled. PXE ROM must support HTTP requests.')
         if self.ipxe and self.http:
             self.filename = 'http://{fileserver}/{filename}'.format(fileserver = self.fileserver, filename = self.filename)
         if self.ipxe and not self.http:
             self.filename = 'tftp://{fileserver}/{filename}'.format(fileserver = self.fileserver, filename = self.filename)
 
-        if self.mode_debug:
-            print 'NOTICE: DHCP server started in debug mode. DHCP server is using the following:'
-            print '\tDHCP Server IP: {}'.format(self.ip)
-            print '\tDHCP Server Port: {}'.format(self.port)
-            print '\tDHCP Lease Range: {} - {}'.format(self.offerfrom, self.offerto)
-            print '\tDHCP Subnet Mask: {}'.format(self.subnetmask)
-            print '\tDHCP Router: {}'.format(self.router)
-            print '\tDHCP DNS Server: {}'.format(self.dnsserver)
-            print '\tDHCP Broadcast Address: {}'.format(self.broadcast)
-            print '\tDHCP File Server IP: {}'.format(self.fileserver)
-            print '\tDHCP File Name: {}'.format(self.filename)
-            print '\tProxyDHCP Mode: {}'.format(self.mode_proxy)
-            print '\tUsing iPXE: {}'.format(self.ipxe)
-            print '\tUsing HTTP Server: {}'.format(self.http)
+        self.logger.debug('NOTICE: DHCP server started in debug mode. DHCP server is using the following:')
+        self.logger.debug('\tDHCP Server IP: ' + self.ip)
+        self.logger.debug('\tDHCP Server Port: ' + str (self.port))
+        self.logger.debug('\tDHCP Lease Range: ' + self.offerfrom + ' - ' + self.offerto)
+        self.logger.debug('\tDHCP Subnet Mask: ' + self.subnetmask)
+        self.logger.debug('\tDHCP Router: ' + self.router)
+        self.logger.debug('\tDHCP DNS Server: ' + self.dnsserver)
+        self.logger.debug('\tDHCP Broadcast Address: ' + self.broadcast)
+        self.logger.debug('\tDHCP File Server IP: ' + self.fileserver)
+        self.logger.debug('\tDHCP File Name: ' + self.filename)
+        self.logger.debug('\tProxyDHCP Mode: ' + str(self.mode_proxy))
+        self.logger.debug('\tUsing iPXE: ' + str(self.ipxe))
+        self.logger.debug('\tUsing HTTP Server: ' + str(self.http))
```

Comment:
I'm not sure about this one, it does improve readability. Here's some test output:
```
2015-04-15 20:44:06,145 INFO Starting TFTP server...
2015-04-15 20:44:06,145 DEBUG TFTP server started in debug mode. TFTP server is using the following:
2015-04-15 20:44:06,145 DEBUG   TFTP Server IP: 0.0.0.0
2015-04-15 20:44:06,145 DEBUG   TFTP Server Port: 79
2015-04-15 20:44:06,146 DEBUG   TFTP Network Boot Directory: .
2015-04-15 19:44:06,146 INFO Starting DHCP server...
2015-04-15 19:44:06,146 DEBUG NOTICE: DHCP server started in debug mode. DHCP server is using the following:
2015-04-15 19:44:06,146 DEBUG   DHCP Server IP: 192.168.2.2
2015-04-15 19:44:06,146 DEBUG   DHCP Server Port: 67
2015-04-15 19:44:06,146 DEBUG   DHCP Lease Range: 192.168.2.100 - 192.168.2.150
2015-04-15 19:44:06,146 DEBUG   DHCP Subnet Mask: 255.255.255.0
2015-04-15 19:44:06,146 DEBUG   DHCP Router: 192.168.2.1
2015-04-15 19:44:06,146 DEBUG   DHCP DNS Server: 8.8.8.8
2015-04-15 19:44:06,146 DEBUG   DHCP Broadcast Address: <broadcast>
2015-04-15 19:44:06,146 DEBUG   DHCP File Server IP: 192.168.2.2
2015-04-15 19:44:06,147 DEBUG   DHCP File Name: pxelinux.0
2015-04-15 19:44:06,147 DEBUG   ProxyDHCP Mode: False
2015-04-15 19:44:06,147 DEBUG   Using iPXE: False
2015-04-15 19:44:06,147 DEBUG   Using HTTP Server: False
2015-04-15 19:44:06,147 INFO PyPXE successfully initialized and running!
```

However, I'm not against removing this.